python/lsst/drp/tasks/assemble_cell_coadd.py

```diff
@@ -340,30 +345,49 @@ def run(self, inputWarps, skyInfo, **kwargs):
-                    ccd_table = (
-                        warp.getInfo()
-                        .getCoaddInputs()
-                        .ccds.subsetContaining(cell_centers_sky[cellInfo.index])
-                    )
-                    assert len(ccd_table) > 0, "No CCD from a warp found within a cell."
-                    assert len(ccd_table) == 1, "More than one CCD from a warp found within a cell."
-                    ccd_row = ccd_table[0]
-                else:
-                    for ccd_row in warp.getInfo().getCoaddInputs().ccds:
-                        if ccd_row.contains(cell_centers_sky[cellInfo.index]):
-                            break
+                ccd_table = (
+                    warp.getInfo().getCoaddInputs().ccds.subsetContaining(cell_centers_sky[cellInfo.index])
+                )
+                assert len(ccd_table) > 0, "No CCD from a warp found within a cell."
+                assert len(ccd_table) == 1, "More than one CCD from a warp found within a cell."
+                ccd_row = ccd_table[0]
 
                 observation_identifier = ObservationIdentifiers.from_data_id(
                     warpRef.dataId,
                     backup_detector=ccd_row["ccd"],
                 )
                 observation_identifiers_gc[cellInfo.index].append(observation_identifier)
 
+                stacker = gc[cellInfo.index]
+                stacker.add_masked_image(mi, weight=weight)
+
+                calexp_point = ccd_row.getWcs().skyToPixel(cell_centers_sky[cellInfo.index])
+                undistorted_psf_im = ccd_row.getPsf().computeImage(calexp_point)
+                # Adjust the bounding box of undistorted_psf_im to be centered
+                # at the closest integer location to the requested location.
+                undistorted_psf_im.setXY0(
+                    geom.Box2I.makeCenteredBox(calexp_point, undistorted_psf_im.getDimensions()).getMin()
+                )
+
+                # Convert the PSF image from Image to MaskedImage.
+                undistorted_psf_maskedImage = afwImage.MaskedImageD(image=undistorted_psf_im)
+                # TODO: In DM-43585, use the variance plane value from noise.
+                undistorted_psf_maskedImage.variance += 1.0  # Set variance to 1
+
+                warped_psf_maskedImage = self.psf_warper.warpImage(
+                    destWcs=skyInfo.wcs,
+                    srcImage=undistorted_psf_maskedImage,
+                    srcWcs=ccd_row.getWcs(),
+                    destBBox=psf_bbox_gc[cellInfo.index],
+                )
+
+                # There may be NaNs in the PSF image. Set them to 0.0
+                warped_psf_maskedImage.variance.array[np.isnan(warped_psf_maskedImage.image.array)] = 1.0
+                warped_psf_maskedImage.image.array[np.isnan(warped_psf_maskedImage.image.array)] = 0.0
+
+                psf_stacker = psf_gc[cellInfo.index]
+                psf_stacker.add_masked_image(warped_psf_maskedImage, weight=weight)
```

Comment:
Are we using a `MaskedImage` here due to limitations in `AccumulatorMeanStacker`?

Reply:
We use `MaskedImage` mainly it warps differently from an `Image` (known issue, unknown cause). I could pick up the `image` plane and call `add_image` here, but given that the conversion is already made, I figured this keeps the code paths for images and PSFs least divergent.